### PR TITLE
Build failure in Travis CI due to connection time out

### PR DESCRIPTION
**Pull request by shinaoka**
Multiple builds in Travis CI sometimes fail due to connection time out (apt & Eigen3).
Here is some workaround for this issue.

### CODE REVIEW

File: common/cmake/ALPSEnableEigen.cmake [hunk 1]
```diff
@@ -74,11 +74,17 @@ function(add_eigen)
       message(STATUS "Trying to download and unpack Eigen3")
       if (NOT EXISTS "${ALPS_EIGEN_TGZ_FILE}")
         message(STATUS "Downloading Eigen3, timeout 600 sec")
-        file(DOWNLOAD ${ALPS_EIGEN_DOWNLOAD_LOCATION} ${ALPS_EIGEN_TGZ_FILE}
-          INACTIVITY_TIMEOUT 60
-          TIMEOUT 600
-          STATUS status_
-          SHOW_PROGRESS)
+        # Attempt to download four times
+        foreach(loop_var RANGE 3)
```
[the remark this answers is not in the document]
Comment (shinaoka) [reply common/cmake/ALPSEnableEigen.cmake]: Thank you for the comment.
Yes, it is better to introduce a parameter.

What will side effects be if the default value of BUNDLE_DOWNLOAD_TRIES is > 1?

[the remark this answers is not in the document]
Comment (mwallerb) [reply common/cmake/ALPSEnableEigen.cmake]: I would just say that I, as a user/developer, would not expect the build system to retry a failed download by default.  Admittedly, that's a little bit of bike-shedding though.

**Comment by shinaoka** [comment]
Thank you, Markus and Emanuel !

**Comment by shinaoka** [comment]
All checks have passed. If Emanuel and Markus are happy with this, could you merge it into the master branch?

**Pull request merged**